src/app/login/login-form.component.ts

```diff
@@ -23,9 +32,17 @@ export class LoginFormComponent {
 
   onSubmit() {
     if (this.createMode) {
-      this.createUser(this.model);
+      if (this.userForm.valid) {
+        this.createUser(this.model);
+      } else {
+        this.message = 'please, fill up required credential';

```

Comment:
this should come from reactive form validator not as separate message.
Also there is not event triggering invalidate message show